PythonVisualizations/Heap.py

```diff
@@ -0,0 +1,518 @@
+            "Random Fill ", lambda: self.clickRandomFill(), numArguments=1,
+            validationCmd=vcmd)
+        removeMaxButton = self.addOperation(
+            "Remove Max", lambda: self.clickRemoveMax())
+        heapifyButton = self.addOperation(
+            "Heapify", lambda: self.clickHeapify())
+        return [insertButton, newHeapButton, peekButton, randomFillButton, removeMaxButton, heapifyButton]
+
+
+    def validArgument(self):
+        entered_text = self.getArgument()
+        if entered_text and entered_text.isdigit():
+            val = int(entered_text)
+            if val < 100:
+                return val
+
+    # Button functions
+    def clickInsert(self):
+        val = self.validArgument()
+        if val is None:
+            self.setMessage("Input value must be an integer from 0 to 99.")
+        else:
+            if len(self.list) == self.MAX_SIZE:
+                self.setMessage("Error! Heap is already full.")
+            else:
+                self.insert(val)
+                self.setMessage("Value {} inserted!".format(val))
+        self.clearArgument()
+
+    def clickPeek(self):
+        if len(self.list) <= 0:
+            self.setMessage("Error! Heap is empty.")
+        else:
+            val = self.list[0].val
+            self.peek()
+            self.setMessage("{} is the root value!".format(val))
+
+
+    def clickNewArray(self):
+        self.newArray()
+
+    def clickRemoveMax(self):
+        if len(self.list) == 0:
+            self.setMessage('Heap is empty!')
+
+        else:
+            val = self.list[0].val
+            self.removeMax()
+            self.setMessage("{} was removed!".format(val))
+
+        self.setButtonsStatus()
+
+
+    def clickRandomFill(self):
+        # if the animation is not stopped (it is running or paused):
+        if self.animationState != self.STOPPED:
+            # error message appears and insert will not take place
+            self.setMessage("Unable to insert at the moment")
+        else:
+            val = self.validArgument()
+            if val is None:
+                self.setMessage("Input value must be an integer from 1 to {}.".format(self.MAX_SIZE))
+            elif self.window.winfo_width() <= self.HEAP_X0 + (
+                    (len(self.list) + 1) * self.CELL_WIDTH):
+                self.setMessage("Error! No room to display")
+            elif 0 < val < 32:
+                self.randomFill(val)
+
+
+    def clickHeapify(self):
+        self.heapify()
+        self.setButtonsStatus()
+
+        # if no button is specified to change its state, all buttons will be set to default state NORMAL, unless a different state is specified
+
+
+    def setButtonsStatus(self, state=NORMAL, buttonName=None):
+        if not buttonName:
+            for b in self.buttons:
+                b['state'] = state
+        else:
+            for b in self.buttons:
+                if b["text"] == buttonName: b['state'] = state

```

Comment:
I think I see what you were intending to do here.  When Random Fill is run, the Insert and Remove Max buttons should be disabled since the heap condition is not satisfiied.  Similarly Heapify should probably only be enabled when the heap condition is unsatisfied.   I suggest changing this method to not take the `state` and `buttonName` parameters, but instead call `isHeap()` and set the button statuses for all buttons accordingly.